ARCHITECTURE.md

```diff
@@ -40,27 +40,174 @@ The Python code in HOOMD-blue is mostly written to wrap core functionality
 written in C++/HIP.  Priority is given to ease of use for users in Python even
 at the cost of code complexity (within reason).
 
+**Note**: Most internal functions, classes, and methods are documented for
+developers (where they are not feel free to add documentation or request it).
+Thus, this section will not go over individual functions or methods in detail
+except where necessary.
+
 _TODO_: Add the basic inheritance and features of HOOMD-blue Python
 objects/operations.
 
+### Base Data Model
+
+#### Definitions
+- added: An object is associated with a simulation
+- attached: An object has its corresponding C++ class instantiated and is
+  connected to a simulation
+- unattached: An object's data resides in pure Python and may be or not be
+  added.
+- type parameter: An attribute that must be specified for all types or groups of
+  types of a set length
+
+In Python, many base classes exist to facilitate code reuse. This leads to very
+lean user facing classes, but requires more fundamental understanding of the model
+to address cases where customization is necessary, or changes are required. The
+first aspect of the data model discussed is that of most HOOMD operations or
+related classes.
+
+- `Operation`
+    - `_HOOMDBaseObject`
+        - `_HOOMDGetSetAttrBase`
+        - `_DependencyRelation`
+
+#### `_DependencyRelation`
+
+`_DependencyRelation` helps define a dependent dependency relationship between
+two objects. The class inherited by `_HOOMDBaseObject` to handle dependent
+relationships between operations in Python. This feature has yet to be used to
```

Comment:
"Yet to be used" is now inaccurate with the recent PRs starting to use it. Describe what it is used for.